contracts/Federation.sol

```diff
@@ -0,0 +1,114 @@
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/ownership/HasNoContracts.sol";
+import "openzeppelin-solidity/contracts/ownership/HasNoTokens.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/// @title Orbs federation smart contract.
+contract Federation is Ownable, HasNoContracts, HasNoTokens {
+    using SafeMath for uint256;
+
+    // The version of the current Federation smart contract.
+    string public constant VERSION = "0.1";
+
+    // Maximum number of the federation members.
+    uint public constant MAX_FEDERATION_MEMBERS = 100;
+
+    // Array of the federations members' public addresses.
+    address[] public members;
+
+    event MemberAdded(address indexed member);
+    event MemberRemoved(address indexed member);
+
+    /// @dev Constructor that initializes the Orbs federation contract.
+    /// @param _members address[] The public addresses of the federation members.
+    constructor(address[] _members) public {
+        require(isFedererationMembersListValid(_members), "Invalid federation members list!");
+
+        members = _members;
+    }
+
+    /// @dev Returns the federation members. Please note that this method is only required due to the current Solidity's
+    /// version inability to support accessing another contract's array using its built-in getter.
+    function getMembers() public view returns (address[]) {
+        return members;
+    }
+
+    /// @dev Adds new member to the federation.
+    /// @param _member address The public address of the new member.
+    function addMember(address _member) public onlyOwner {
+        require(_member != address(0), "Address must not be 0!");
+        require(members.length + 1 <= MAX_FEDERATION_MEMBERS, "Can't add more members!");
+
+        // Check for duplicates.
+        for (uint i = 0; i < members.length; ++i) {
+            require(members[i] != _member, "Can't add a duplicate member!");
+        }
+
+        members.push(_member);
+        emit MemberAdded(_member);
+    }
+
+    /// @dev Removes existing member from the federation.
+    /// @param _member address The public address of the existing member.
+    function removeMember(address _member) public onlyOwner {
+        require(_member != address(0), "Address must not be 0!");
+        require(members.length - 1 > 0, "Can't remove all members!");
+
+        // Check for existence.
+        (uint i, bool exists) = findMemberIndex(_member);
+        require(exists, "Member doesn't exist!");
+
+        removeMemberByIndex(i);
+        emit MemberRemoved(_member);
+    }
+
+    /// @dev Returns an index of an existing member. Returns whether the member exist.
+    /// @param _member address The public address of the member to look for.
+    function findMemberIndex(address _member) private view returns(uint, bool) {
+        uint i;
+        for (i = 0; i < members.length; ++i) {
+            if (members[i] == _member) {
```

Comment:
Don't you get the same issue we talked about in the past that the last member is not scanned?

Reply:
I think that it's not the case, because last time we were checking duplicates by pairs, while in this case we just enumerate the array and compare them one by one.